libr/io/map.c

```diff
@@ -60,8 +60,16 @@ static bool _map_skyline_push(RPVector *map_skyline, ut64 from, ut64 to, RIOMap
 	return true;
 }
 
-// Store map parts that are not covered by others into io->map_skyline
-R_API void r_io_map_calculate_skyline(RIO *io) {
+/**
+ * Update the internal state of IO. It store map parts that are not "shadowed"
+ * by others, so that only the visible map parts can be accessed.
+ *
+ * NOTE: you shouldn't need this function in the general case, because the state
+ *       should always be consistent unless you use `_batch` functions. If you
+ *       do use `_batch` functions, be sure to call `r_io_update` when you are
+ *       done, to ensure IO state consistency.
+ */
+R_API void r_io_update(RIO *io) {
```

Comment:
Since this function is warned to discourage normal usage. It can be clearer to be named `r_io_update_skyline`

Reply:
I didn’t want to tie the name to the implementation details.